### PR TITLE
Redis add hgetall hdel

### PR DESCRIPTION
**Pull request by lantosgyuri**

### CODE REVIEW

File: pkg/redis/client_test.go [hunk 1]
```diff
@@ -155,6 +155,17 @@ func (s *ClientWithMiniRedisTestSuite) TestHMGet() {
...
 }
 
+func (s *ClientWithMiniRedisTestSuite) TestHGetAll() {
+	err := s.client.HSet(context.Background(), "key", "field", "value1")
+	s.NoError(err, "there should be no error on HSet")
+	serr := s.client.HSet(context.Background(), "key", "field2", "value2")
+	s.NoError(serr, "there should be no error on HSet")
+
+	vals, err := s.client.HGetAll(context.Background(), "key")
+	s.NoError(err, "there should be no error on HgetAll")
```
Comment (selm0) [inline pkg/redis/client_test.go:165]: `HGetAll` (uppercase G), as that's the command name

File: pkg/redis/client_test.go [hunk 2]
```diff
@@ -155,6 +155,17 @@ func (s *ClientWithMiniRedisTestSuite) TestHMGet() {
...
+	s.Equal(map[string]string{"field": "value1", "field2": "value2"}, vals, "there should be no error on HSet")
```
Comment (selm0) [inline pkg/redis/client_test.go:166]: You're message is misleading, as the assert does not have anything to do with `errors on HSet`.
You can either fix it or remove it.
For most assertions, we don't provide error messages, as they are just duplication the assertion itself:
e.g. s.NoError(error, "there should be no error") contains the same information twice.

File: pkg/redis/client.go [hunk 3]
```diff
@@ -259,6 +262,22 @@ func (c *redisClient) HSet(ctx context.Context, key, field string, value interfa
 	return err
 }
 
+func (c *redisClient) HDel(ctx context.Context, key string, fields ...string) (int64, error) {
```
Comment (selm0) [inline pkg/redis/client.go:265]: Can you please add a test for HDel aswell?

**Review by selm0 — CHANGES_REQUESTED** [review]

**Comment by j4k4** [comment]
do we have 2 PRs now for this? :thinking:

File: pkg/redis/client.go [hunk 4]
```diff
@@ -47,6 +47,8 @@ type Client interface {
 	HMGet(ctx context.Context, key string, fields ...string) ([]interface{}, error)
 	HMSet(ctx context.Context, key string, pairs map[string]interface{}) error
 	HSetNX(ctx context.Context, key string, field string, value interface{}) (bool, error)
+	HDel(ctx context.Context, key string, fields ...string) (int64, error)
```
Comment (selm0) [inline pkg/redis/client.go:50]: Please order all the methods alphabetically

**Review by selm0 — CHANGES_REQUESTED** [review]

**Review by selm0 — APPROVED** [review]

**Pull request merged**